opc/resource_machine_image.go

```diff
@@ -0,0 +1,203 @@
+package opc
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-oracle-terraform/client"
+	"github.com/hashicorp/go-oracle-terraform/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/structure"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceOPCMachineImage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOPCMachineImageCreate,
+		Read:   resourceOPCMachineImageRead,
+		Delete: resourceOPCMachineImageDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateComputeStorageAccountName,
+			},
+
+			"attributes": {
+				Type:             schema.TypeString,
+				ForceNew:         true,
+				Optional:         true,
+				ValidateFunc:     validation.ValidateJsonString,
+				DiffSuppressFunc: structure.SuppressJsonDiff,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"error_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hypervisor": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+
+			"image_format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"file": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"no_upload": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"platform": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			// TODO
```

Comment:
Can we put a little blurb about why this is TODO? Just so we know when to come back to it

Reply:
None of the machine images I checked returned a sizes response, so couldn't validate the data structure.  I'll double check again and either update or remove.

Reply:
Fixed (removed)